spacy_experimental/span_finder/span_finder_component.py

```diff
@@ -0,0 +1,323 @@
+    ) -> None:
+        """Initialize the span boundary detector.
+        model (thinc.api.Model): The Thinc Model powering the pipeline component.
+        name (str): The component instance name, used to add entries to the
+            losses during training.
+        threshold (float): Minimum probability to consider a prediction
+            positive.
+        scorer (Optional[Callable]): The scoring method.
+        candidates_key (str): Name of the SpanGroup the span candidates are saved to
+        max_length (int): Max length of the produced spans (unlimited when set to 0)
+        min_length (int): Min length of the produced spans (unlimited when set to 0)
+        """
+        self.vocab = nlp
+        self.cfg = {
+            "threshold": threshold,
+            "max_length": max_length,
+            "min_length": min_length,
+            "candidates_key": candidates_key,
+        }
+        self.model = model
+        self.name = name
+        self.scorer = scorer
+
+    def predict(self, docs: Iterable[Doc]):
+        """Apply the pipeline's model to a batch of docs, without modifying them.
+        docs (Iterable[Doc]): The documents to predict.
+        RETURNS: The models prediction for each document.
+        """
+        scores = self.model.predict(docs)
+        return scores
+
+    def set_annotations(self, docs: Iterable[Doc], scores: Floats2d) -> None:
+        """Modify a batch of Doc objects, using pre-computed scores.
+        docs (Iterable[Doc]): The documents to modify.
+        scores: The scores to set, produced by SpanFinder predict method.
+        """
+        lengths = [len(doc) for doc in docs]
+
+        offset = 0
+        scores_per_doc = []
+        for length in lengths:
+            scores_per_doc.append(scores[offset : offset + length])
+            offset += length
+
+        for doc, doc_scores in zip(docs, scores_per_doc):
+            doc.spans[self.cfg["candidates_key"]] = []
+            starts = []
+            ends = []
+
+            for token, token_score in zip(doc, doc_scores):
+                if token_score[0] >= self.cfg["threshold"]:
+                    starts.append(token.i)
+                if token_score[1] >= self.cfg["threshold"]:
+                    ends.append(token.i)
+
+            for start in starts:
+                for end in ends:
+                    if start < end + 1:
+                        span_length = end + 1 - start
+                        if (
+                            self.cfg["min_length"] <= 0
+                            or span_length >= self.cfg["min_length"]
+                        ) and (
+                            self.cfg["max_length"] <= 0
+                            or span_length <= self.cfg["max_length"]
+                        ):
+                            doc.spans[self.cfg["candidates_key"]].append(
+                                doc[start : end + 1]
+                            )
```

Comment:
```suggestion
                    span_length = end + 1 - start
                    if span_length > 0:
                        if (
                            self.cfg["min_length"] <= 0
                            or span_length >= self.cfg["min_length"]
                        ) and (
                            self.cfg["max_length"] <= 0
                            or span_length <= self.cfg["max_length"]
                        ):
                            doc.spans[self.cfg["candidates_key"]].append(
                                doc[start : end + 1]
                            )
                        elif (
                            self.cfg["max_length"] > 0
                            and span_length > self.cfg["max_length"]
                        ):
                            break
```

Reply:
Maybe you can add a unit test where you test just `set_annotations` with mock scores?